pontoon/base/models.py

```diff
@@ -2245,6 +2248,26 @@ def reset_active_translations(self, locale):
 
         translations.filter(pk__in=unreviewed_pks).update(active=True)
 
+    def _get_word_count(self, string):
+        return len(re.findall(r'\w+', string))
+
+    def get_or_create(self, defaults=None, **kwargs):
+        if 'string' in kwargs:
+            kwargs['word_count'] = self._get_word_count(kwargs['string'])
+        return super(EntityQuerySet, self).get_or_create(defaults, **kwargs)
+
+    def bulk_update(self, objs, fields=None, batch_size=None):
+        if django.VERSION[0] >= 2:
+            msg = "Django version is 2 or higher. Function bulk_update need to be removed"
+            warnings.warn(msg, PendingDeprecationWarning)
+        if objs:
+            for obj in objs:
+                if hasattr(obj, 'string'):
+                    obj.word_count = self._get_word_count(obj.string)
+                    obj.save()
```

Comment:
Why do we need the two `if` clauses?

You shouldn't `save()` each item individually (slow) and instead let bulk_update make the change in the DB.

Reply:
@mathjazz Save() removed.

> Why do we need the two if clauses?
First clause check that objs exist and second that obj has string attribute. Are they excessive?

Reply:
Good opint - the first one should stay. The second should always be `True`:
https://github.com/mozilla/pontoon/blob/cc3c3267e8f279f1724d1f79fa0d351cb1c544fb/pontoon/sync/changeset.py#L181